pg-extern-attr/src/lib.rs

```diff
@@ -285,11 +281,25 @@ fn impl_info_for_fn(item: &syn::Item) -> TokenStream {
                     result.into_datum()
                 }
                 Err(err) => {
+                    use std::sync::atomic::compiler_fence;
+                    use std::sync::atomic::Ordering;
+
                     // ensure the return value is null
                     func_info.isnull = pg_extend::pg_bool::Bool::from(true).into();
 
-                    // TODO: anything else to cean up before resuming the panic?
-                    panic::resume_unwind(err)
+                    // The Rust code paniced, we need to recover to Postgres via a longjump
+                    //   A postgres logging error of Error will do this for us.
+                    compiler_fence(Ordering::SeqCst);
+                    let level = pg_extend::pg_error::Level::Error;
+                    pg_extend::pg_error::log(
+                        level,
+                        file!(),
+                        line!(),
+                        module_path!(),
+                        format!("panic executing Rust function: {}", stringify!(#func_name)),
```

Comment:
I think we could possibly try and downcast the err to `&'static str` to possibly produce a better message here.